src/pages/staking-hub/nodeAdded.tsx

```diff
@@ -251,9 +254,52 @@ const GrayCard = ({
 };
 
 const NodeAdded = () => {
+  const dispatch = useAppDispatch();
   const nodeNativeAddress = useAppSelector((store) => store.node.addresses.data.native);
   const nodeHoprAddress = useAppSelector((store) => store.node.addresses.data.hopr);
   const safeBalance = useAppSelector((store) => store.safe.balance.data);
+  const loginData = useAppSelector((store) => store.auth.loginData);
+  const nodeSettings = useAppSelector((store) => store.node.settings.data);
+  const wxHoprAllowance = useAppSelector((store) => store.stakingHub.safeInfo.data.allowance.wxHoprAllowance);
+  const safeAddress = useAppSelector((store) => store.safe.selectedSafeAddress.data);
+
+  const signer = useEthersSigner();
+
+  const navigate = useNavigate();
+  const [queryParams, set_queryParams] = useState('');
+
+  useEffect(() => {
+    if (loginData.apiEndpoint && loginData.apiToken) {
+      const queryParams = new URLSearchParams({
+        apiToken: loginData.apiToken,
+        apiEndpoint: loginData.apiEndpoint,
+      }).toString();
+      set_queryParams(queryParams);
+    }
+    handleRefresh();
+  }, [loginData]);
+
+  useEffect(() => {
+    if (signer && safeAddress) {
+      dispatch(
+        safeActionsAsync.getSafeInfoThunk({
+          signer: signer,
+          safeAddress,
+        }),
+      );
+    }
+  }, [safeAddress]);
```

Comment:
This is done already on the ConnectSafe component, why are we doing it again here?